docs/sources/reference/components/prometheus.exporter.snowflake.md

```diff
@@ -11,27 +11,60 @@ The `prometheus.exporter.snowflake` component embeds
 
 ## Usage
 
+### Password Authentication
+
+```alloy
+prometheus.exporter.snowflake "LABEL" {
+    account_name = <ACCOUNT_NAME>
+    username =     <USERNAME>
+    password =     <PASSWORD>
+    warehouse =    <WAREHOUSE>
+}
+```
+
+Replace the following:
+
+- _`<ACCOUNT_NAME>`_: The Snowflake account name you are collecting metrics from.
+- _`<USERNAME>`_: The username used to query metrics.
+- _`<PASSWORD>`_: The password for the user used to query metrics.
+- _`<WAREHOUSE>`_: The virtual warehouse to use when querying metrics.
+
+### RSA Authentication
+
 ```alloy
 prometheus.exporter.snowflake "LABEL" {
-    account_name = ACCOUNT_NAME
-    username =     USERNAME
-    password =     PASSWORD
-    warehouse =    WAREHOUSE
+    account_name =         <ACCOUNT_NAME>
+    username =             <USERNAME>
+    private_key_path =     <PRIVATE_KEY_PATH>
+    private_key_password = <PRIVATE_KEY_PASSWORD>
+    warehouse =            <WAREHOUSE>
 }
 ```
 
+Replace the following:
+
+- _`<ACCOUNT_NAME>`_: The Snowflake account name you are collecting metrics from.
+- _`<USERNAME>`_: The username used to query metrics.
+- _`<PRIVATE_KEY_PATH>`_: The path to the user's RSA private key file.
+- _`<PRIVATE_KEY_PASSWORD>`_: The password for the user's RSA private key.
+- _`<WAREHOUSE>`_: The virtual warehouse to use when querying metrics.
+
 ## Arguments
 
 The following arguments can be used to configure the exporter's behavior.
 Omitted fields take their default values.
-
-| Name           | Type     | Description                                           | Default          | Required |
-| -------------- | -------- | ----------------------------------------------------- | ---------------- | -------- |
-| `account_name` | `string` | The account to collect metrics for.                   |                  | yes      |
-| `username`     | `string` | The username for the user used when querying metrics. |                  | yes      |
-| `password`     | `secret` | The password for the user used when querying metrics. |                  | yes      |
-| `role`         | `string` | The role to use when querying metrics.                | `"ACCOUNTADMIN"` | no       |
-| `warehouse`    | `string` | The warehouse to use when querying metrics.           |                  | yes      |
+One of `password` or `private_key_path` must be specified to authenticate.
```

Comment:
Is the mutual exclusion handled by the underlying library? It would be better if we handled this via the Validate interface for better error messaging.

Reply:
@mattdurham The exporter is handling this case (only erroring on `Validate` if both are not supplied)